### ISSUE TITLE
Profile's name should be unique

### ISSUE THREAD

**Issue by ThisIsClark** [opensds/opensds#987]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
Profile should be unique,  so we should not have two profiles with same name.
Actually, we can not create two profiles with same name and same type, but we can **create two profiles with same name and different types**, which is unreasonable.

**What you expected to happen**:
Can not create two profiles with same name, even though they have different types. 

**How to reproduce it (as minimally and precisely as possible)**:
Create a profile with name A and type B.
Then create another profile with name A and type C, this profile would be created successfully.

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version: latest
- OS (e.g. from /etc/os-release): Ubuntu 16.04.2 LTS (Xenial Xerus)
- Kernel (e.g. `uname -a`): Linux ubuntu 4.4.0-62-generic #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Others:

**Comment by kumarashit** [comment]
Hi, Did you try to create profile using Dashboard? or some other mode cli/API?
I just tried to create profile with same name but different Storage Type and it gives 'Name is existing' error
![DuplicateProfile1](https://user-images.githubusercontent.com/48085413/63424226-98573280-c42b-11e9-8edb-f3fd59307197.PNG)
![DuplicateProfile2](https://user-images.githubusercontent.com/48085413/63424228-98573280-c42b-11e9-9051-b8f08ec1628e.PNG)

**Comment by himanshuvar** [comment]
@kumarashit Using CLI, I tried creating profile with same name but different Storage Type, it allowed me to create the profiles. However, I found another issue that there is no check/validation for the accessprotocol. I tried some random wrong combination for profile creation.
![1](https://user-images.githubusercontent.com/8712951/63428221-a198cd00-c434-11e9-993c-dbcfdce5c573.JPG)
![2](https://user-images.githubusercontent.com/8712951/63428278-cf7e1180-c434-11e9-82c0-d55a42911e2b.JPG)
